res/controllers/Hercules-DJControl-Inpulse-200-script.js

```diff
@@ -0,0 +1,182 @@
+//                       USER OPTIONS                        //
+///////////////////////////////////////////////////////////////
+
+// How fast scratching is.
+DJCi200.scratchScale = 1.0;
+
+// How much faster seeking (shift+scratch) is than scratching.
+DJCi200.scratchShiftMultiplier = 4;
+
+// How fast bending is.
+DJCi200.bendScale = 1.0;
+
+// Other scratch related options
+DJCi200.kScratchActionNone = 0;
+DJCi200.kScratchActionScratch = 1;
+DJCi200.kScratchActionSeek = 2;
+DJCi200.kScratchActionBend = 3;
+
+DJCi200.init = function() {
+    // Scratch button state
+    DJCi200.scratchButtonState = true;
+    // Scratch Action
+    DJCi200.scratchAction = {
+        1: DJCi200.kScratchActionNone,
+        2: DJCi200.kScratchActionNone
+    };
+
+    //Turn On Vinyl buttons LED(one for each deck).
+    midi.sendShortMsg(0x91, 0x03, 0x7F);
+    midi.sendShortMsg(0x92, 0x03, 0x7F);
+
+    //Turn On Browser button LED
+    midi.sendShortMsg(0x90, 0x04, 0x05);
+
+    //Softtakeover for Pitch fader
+    engine.softTakeover("[Channel1]", "rate", true);
+    engine.softTakeover("[Channel2]", "rate", true);
+    engine.softTakeoverIgnoreNextValue("[Channel1]", "rate");
+    engine.softTakeoverIgnoreNextValue("[Channel2]", "rate");
+
+    //Set effects Levels - Dry/Wet
+    engine.setParameter("[EffectRack1_EffectUnit1_Effect1]", "meta", 0.6);
+    engine.setParameter("[EffectRack1_EffectUnit1_Effect2]", "meta", 0.6);
+    engine.setParameter("[EffectRack1_EffectUnit1_Effect3]", "meta", 0.6);
+    engine.setParameter("[EffectRack1_EffectUnit2_Effect1]", "meta", 0.6);
+    engine.setParameter("[EffectRack1_EffectUnit2_Effect2]", "meta", 0.6);
+    engine.setParameter("[EffectRack1_EffectUnit2_Effect3]", "meta", 0.6);
+    engine.setParameter("[EffectRack1_EffectUnit1]", "mix", 1);
+    engine.setParameter("[EffectRack1_EffectUnit2]", "mix", 1);
+
+    // Ask the controller to send all current knob/slider values over MIDI, which will update
+    // the corresponding GUI controls in MIXXX.
+    midi.sendShortMsg(0xB0, 0x7F, 0x7F);
+};
+
+// The Vinyl button, used to enable or disable scratching on the jog wheels (One per deck).
+DJCi200.vinylButton = function(_channel, _control, value, status, _group) {
+    if (value) {
+        if (DJCi200.scratchButtonState) {
+            DJCi200.scratchButtonState = false;
+            midi.sendShortMsg(status, 0x03, 0x00);
+        } else {
+            DJCi200.scratchButtonState = true;
+            midi.sendShortMsg(status, 0x03, 0x7F);
+        }
+    }
+};
+
+DJCi200._scratchEnable = function(deck) {
+    var alpha = 1.0/8;
+    var beta = alpha/32;
+    engine.scratchEnable(deck, 248, 33 + 1/3, alpha, beta);
+};
+
+DJCi200._convertWheelRotation = function(value) {
+    // When you rotate the jogwheel, the controller always sends either 0x1
+    // (clockwise) or 0x7F (counter clockwise). 0x1 should map to 1, 0x7F
+    // should map to -1 (IOW it's 7-bit signed).
```

Comment:
Can you confirm this is the case at higher speeds as well? Controllers usually bundle messages together to not oversaturate the bus. The resulting packets then carry a higher value with them.

Reply:
yes, this is the case for our (Hercules) controllers. Included a capture of scratching with the Inpulse200.
[Inpulse200_Jog_capture.txt](https://github.com/mixxxdj/mixxx/files/4771305/Inpulse200_Jog_capture.txt)